diff --git a/meson.build b/meson.build
index 6a48cbf..fcc49e1 100644
--- a/meson.build
+++ b/meson.build
@@ -169,45 +169,7 @@ install_data(
   install_mode: 'rw-r--r--',
 )
 
-dep_cmocka = dependency('cmocka')
 
-foreach test : ['list', 'navigator']
-  test(
-    'test_@0@'.format(test),
-    executable(
-      'test_@0@'.format(test),
-      [files('test/@0@.c'.format(test), 'src/dummy_window.c'), files_common],
-      include_directories: include_directories('src'),
-      dependencies: [deps_for_imv, dep_cmocka],
-    )
-  )
-endforeach
-
-prog_a2x = find_program('a2x')
-
-foreach man : [
-  [1, 'imv'],
-  [1, 'imv-msg'],
-  [5, 'imv'],
-]
-  _section = man[0]
-  _topic = man[1]
-  custom_target(
-    '@0@(@1@)'.format(_topic, _section),
-    input: 'doc/@0@.@1@.txt'.format(_topic, _section),
-    output: '@0@.@1@'.format(_topic, _section),
-    command: [
-      prog_a2x,
-      '--no-xmllint',
-      '--doctype', 'manpage',
-      '--format', 'manpage',
-      '--destination-dir', meson.current_build_dir(),
-      '@INPUT@'
-    ],
-    install: true,
-    install_dir: '@0@/man@1@'.format(get_option('mandir'), _section)
-  )
-endforeach
 
 message('\n\n'
 + 'Building imv @0@\n\n'.format(meson.project_version())
