diff --git a/src/compat-api.h b/src/compat-api.h
index d09e1fb..b0ede03 100644
--- a/src/compat-api.h
+++ b/src/compat-api.h
@@ -224,3 +224,7 @@ static inline void FreePixmap(PixmapPtr pixmap)
 #endif
 
 #endif
+
+#if XORG_VERSION_CURRENT >= XORG_VERSION_NUMERIC(1,17,99,0,0)
+#define PixmapSyncDirtyHelper(d, dd) PixmapSyncDirtyHelper(d)
+#endif
diff --git a/src/sna/sna_driver.c b/src/sna/sna_driver.c
index 8a3599c..48b927b 100644
--- a/src/sna/sna_driver.c
+++ b/src/sna/sna_driver.c
@@ -57,6 +57,8 @@ USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <mi.h>
 #include <micmap.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
 #include <sys/ioctl.h>
 #include <sys/fcntl.h>
 #include <sys/poll.h>
diff --git a/src/uxa/intel_driver.c b/src/uxa/intel_driver.c
index 2793da5..945b8bd 100644
--- a/src/uxa/intel_driver.c
+++ b/src/uxa/intel_driver.c
@@ -45,6 +45,10 @@ USE OR OTHER DEALINGS IN THE SOFTWARE.
 #include <stdio.h>
 #include <errno.h>
 
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
 #include "xorg-server.h"
 #include "xf86.h"
 #include "xf86_OSproc.h"
